docs/derivation-rules.saty

```diff
@@ -0,0 +1,202 @@
+        ];
+      }
+    >
+    +p{
+      Here are the derivation rules:
+    }
+    +small-block<
+     +math(
+       ${
+         \derive?:{\paren{\mathsc{VAR}}}{
+           | |
+         }{
+           \judgement{A \cup \{x \mapsto \tau\}}{x}{\tau}{\emptyset}
+         }
+       }
+     );
+    >
+    +small-block<
+      +math(
+        ${
+          \derive?:{\paren{\mathsc{STRUCT}}}{
+            | \judgement{A}{e_1}{\tau_1}{C_1} | \cdots | \judgement{A}{e_n}{\tau_n}{C_n} |
+          }{
+            \judgement{A}{\{e_1, \cdots , e_n\}}{\{\tau_1, \cdots , \tau_n\}}{C_1 \wedge \cdots \wedge C_n}
+          }
+        }
+      );
+    >
+    +small-block<
+      +math(
+        ${
+          \derive?:{\paren{\mathsc{LET}}}{
+            | \judgement{A}{e_1}{\tau_1}{C_1} | \judgement{A \cup \{x \mapsto \tau_1\}}{e_2}{\tau_2}{C_2} |
+          }{
+            \judgement{A}{\mathtt{let}\ x = e_1\ \mathtt{in}\ e_2}{\tau_2}{C_1 \wedge C_2}
+          }
+        }
+      );
+    >
+    +small-block<
+      +math(
+        ${
+          \derive?:{\paren{\mathsc{LETREC}}}{
+            | \judgement{A'}{fn_1}{\tau_1}{C_1} \cdots
+              \judgement{A'}{fn_n}{\tau_n}{C_n} |
+              \judgement{A'}{e}{\tau}{C} |
+              \mathtt{where}\ A' = A \cup \{x_i \mapsto \alpha_i\} |
+          }{
+            \judgement{A}{\mathtt{letrec}\ x_1 = f_1, \cdots , x_n = f_n\ \mathtt{in}\ e}{\tau}{C_1 \wedge \cdots \wedge C_n \wedge C \wedge \(\tau'_1 = \tau_1\) \wedge \cdots \wedge \(\tau'_n = \tau_n\)}
+          }
+        }
+      );
+    >
+    +small-block<
+      +math(
+        ${
+          \derive?:{\paren{\mathsc{ABS}}}{
+            | \judgement{A \cup \{x_1 \mapsto \alpha_1, \cdots , x_n \mapsto \alpha_n\}}{e}{\tau}{C} |
+          }{
+            \judgement{A}{\mathtt{fun}\(x_1, \cdots , x_n\) \rightarrow e}{\(\alpha_1, \cdots , \alpha_n\) \rightarrow \tau}{C}
+          }
+        }
+      );
+    >
+    +small-block<
+      +math(
+        ${
+          \derive?:{\paren{\mathsc{APP}}}{
+            | \judgement{A}{e}{\tau}{C} |
+              \judgement{A}{e_1}{\tau_1}{C_1} \cdots \judgement{A}{e_n}{\tau_n}{C_n} |
+          }{
+            \judgement{A}{e\(e_1, \cdots , e_n\)}{\beta}{\(\tau = \(\alpha_1, \cdots , \alpha_n\) \rightarrow \alpha\) \wedge \(\beta \subseteq \alpha\) \wedge \(\tau_1 \subseteq \alpha_1\) \wedge \cdots \wedge \(\tau_n \subseteq \alpha_n\) \wedge C \wedge C_1 \wedge \cdots \wedge C_n}
+          }
+        }
+      );
+    >
+    +small-block<
+      +math(
+        ${
+          \derive?:{\paren{\mathsc{PAT}}}{
+            | \judgement{A}{p}{\tau}{C_p} |
+              \judgement{A}{g}{\tau_g}{C_g} |
+          }{
+            \judgement{A}{p\ \mathtt{when}\ g}{\tau}{\(\tau \subseteq \mathtt{boolean\(\)}\) \wedge C_p \wedge C_g}
+          }
+        }
+      );
+    >
+    +small-block<
+      +math(
+        ${
+          \derive?:{\paren{\mathsc{CASE}}}{
+            | \judgement{A}{p}{\tau}{C_e} |
```

Comment:
Is it a typo?
```suggestion
            | \judgement{A}{e}{\tau}{C_e} |
```

Reply:
It is a typo! Thanks!

Reply:
fixed https://github.com/dwango/fialyzer/pull/205/commits/551ebc7d6d811bc4047255f97e49b407a8d36f80